### PR TITLE
Instances for `PlutusTx.Rational` and `NonEmpty`

### PR DESCRIPTION
**Pull request by sergesku**
- Instances of `Eq`, `Ord`, `Arbitrary`, `CoArbitrary`
  and `Function` for `NonEmpty`
- Instances of `Arbitrary`, `CoArbitrary` and `Function` for
  `PlutusTx.Ratio.Rational`

### CODE REVIEW

File: plutus-extra/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 This format is based on [Keep A Changelog](https://keepachangelog.com/en/1.0.0).
 
+## 3.1 --
```
Comment (kozross) [inline plutus-extra/CHANGELOG.md:5]: Needs a date.

File: plutus-extra/src/PlutusTx/NonEmpty.hs [hunk 2]
```diff
@@ -66,10 +72,16 @@ infixr 5 :|
...
-
-instance Show a => Show (NonEmpty a) where
+  deriving stock
+    ( -- | @since 3.1
+      Prelude.Eq
+    , -- | @since 3.1
+      Prelude.Ord
+    )
+
+instance Prelude.Show a => Prelude.Show (NonEmpty a) where
```
Comment (kozross) [inline plutus-extra/src/PlutusTx/NonEmpty.hs:82]: We can probably just stock-derive this.

**Review by kozross — CHANGES_REQUESTED** [review]

File: plutus-extra/src/PlutusTx/NonEmpty.hs (hunk 2 above)
Comment (sergesku) [reply plutus-extra/src/PlutusTx/NonEmpty.hs:82]: The idea was
```
>>> singleton True
NonEmpty [True]
```
Do we need it?

Comment (kozross) [reply plutus-extra/src/PlutusTx/NonEmpty.hs:82]: Not really. `Show` is only for debugging really, and it's usually best to stock-derive it, due to things like `pretty-show`.

**Review by kozross — APPROVED** [review]

**Pull request merged**